### PR TITLE
Endpoint to delete existing repo

### LINKED ISSUE #64: Remove franklin repo
Endpoint to delete all data related to a repo.

### PR DESCRIPTION
**Pull request by oudeismetis**
fixes #64

### CODE REVIEW

File: franklin/core/urls.py [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
 from github.views import deploy_hook, deployed_repos, deployable_repos, \
-    register_repo, get_auth_token
+    repository, get_auth_token
 
 urlpatterns = [
     url(r'^auth/github/$', get_auth_token, name='get_token'),
     url(r'^dashboard/init/$', dashboard_init, name='dashboard_init'),
     url(r'^deployed/$', deploy_hook, name='deploy'),
-    url(r'^register/$', register_repo, name='register'),
+    url(r'^repo/$', repository, name='repository'),
```
Comment (oudeismetis) [inline franklin/core/urls.py]: @JulianGindi @CorneliusIV @camiriveron breaking change here.

File: franklin/github/views.py [hunk 2]
```diff
@@ -111,10 +122,13 @@ def register_repo(request):
                     return Response(status=response.status_code)
                 else:
                     return Response(status=status.HTTP_201_CREATED)
-            else:
-                message = 'Current user does not have deployment permissions'
-                logger.warn(message + ' | %s | %s', request.user, site)
-                return Response(message, status=status.HTTP_403_FORBIDDEN)
+            if request.method == 'DELETE':
+                site.delete()
+                return Response(status=status.HTTP_204_NO_CONTENT)
```
Comment (oudeismetis) [inline franklin/github/views.py]: piggybacking on the existing `/registration` endpoint gets us a lot for free here. Plus one for being RESTful.

File: franklin/github/views.py [hunk 3]
```diff
@@ -111,10 +122,13 @@ def register_repo(request):
                     return Response(status=response.status_code)
                 else:
                     return Response(status=status.HTTP_201_CREATED)
-            else:
-                message = 'Current user does not have deployment permissions'
-                logger.warn(message + ' | %s | %s', request.user, site)
-                return Response(message, status=status.HTTP_403_FORBIDDEN)
+            if request.method == 'DELETE':
```
Comment (camiriveron) [inline franklin/github/views.py]: if - else if increases performance, right?

**Comment by JulianGindi** [comment]
:+1:

File: franklin/github/views.py (hunk 3 above)
Comment (oudeismetis) [inline franklin/github/views.py]: ever so slightly, yes :)

**Pull request merged**